[WIP] Documentation FAQ & General Improvements

Still34 opened the pull request:
# Preview

[![Build status](https://ci.appveyor.com/api/projects/status/hq756u9qssm6a2xs?svg=true)](https://ci.appveyor.com/project/Still/discord-net)

The latest preview can be seen [here](https://docs.stillu.cc). 

It has been set up with AppVeyor to be in sync with the commits from the PR branch.

# Purpose

This PR aims to do the following,

1. Improve the XMLDocs in many sections of the library. This includes classes, props, methods, etc.
2. Add a FAQ section under the current documentation page.
3. Improve the general documentations.
4. General cleanup (add `ConfigureAwait(false)`; remove unused references) in the codebase.

This PR is currently under work-in-progress. Any suggestions are welcome.

# To-do

- [ ] Finish covering common entities' XMLDocs
    - This includes at least the base interfaces and classes.
- [ ] Finish building FAQ
    - [ ] Add "I want to build a bot with a UI" section.
    - [ ] Add "Entity Comparison" section.
- [x] Proofread guide to ensure that it is up-to-date
- [x] Enable `View Source` button
    - Added in 6f4a4e0.
    * Current resolution in the commit is not ideal, please check the commit detail to see more regarding this issue.
- [x] Find a workaround for constantly changing method signature names causing xref to fail
    - Resolved in 291ad1cf. 
    - Method names will require a wildcard symbol at the end of the full name to redirect to the proper method. e.g. ``Discord.Commands.ModuleBase`1.ReplyAsync*``
    - Event and class names do not require any wildcard symbol.
- [x] Enable site-wide search via docfx
   - Resolved in 68be9d3.
   - Searching in documentation for a method was often difficult, or had to rely on separate service such as Googling the host (e.g. `SendMessageAsync site:discord.foxbot.me`) or .NET Discord bot. The built-in search function for docfx should help aid this.
   - This was enabled via `_enableSearch` under `docfx.json`.
- [x] Fix foreign cross-referencing
   - Resolved in 8be767b6.
   - An external cross-referencing service was required for this fix. Our `docfx.json` now references MSDN xref service.
   - This means objects that used to render `@System.IServiceProvider` now renders into `[IServiceProvider](https://docs.microsoft.com/dotnet/api/system.iserviceprovider)` automatically.
- [x] Create a dark template
   - Initial support added in 83435aa.
   - Theme switcher added in 6511b1a.

Commit:
Fix broken cross references

Commit:
Add FAQ section

Commit:
Add dev mode image

Commit:
Remove migration docs

- The link never worked in the first place and is rarely used.

Commit:
Comply with CONTRIBUTING.md

- As well as making minor adjustments here and there

Commit:
Swap Guides and FAQ position

Commit:
Add more glossary types

Commit:
Add channel type anchor

Commit:
Rename filenames and remove adv op

Commit:
Sort order

Commit:
Add xmldocs for CommandExecuted and various methods

Commit:
Add RunMode documentation

Commit:
Grammar fix

Commit:
Remove unnecessary RunMode reference

Commit:
Add CommandError xmldocs

Commit:
Add ModuleBase and EmbedBuilder docfx

Commit:
Add "How can I get the guild from a message?"

Commit:
Change wording with hyperlink section

Commit:
Fix typo & improve styling in Basic Operations

Commit:
Add examples & tidy commands section

+ Modified the headings to fit the formal aspect of docs.
+ Added more details regarding RunMode

Commit:
Add self-bot warning & clean-up

Joe4evr commented on docs/faq/Commands.md:
```diff
@@ -0,0 +1,142 @@
+[!code-csharp[DI](samples/commands/DI.cs)]
+
+[Dependency Injection]: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection
+
```
Consider adding a link to [this video](https://www.youtube.com/watch?v=QtDTfn8YxXg) as well.

Still34 replied:
Added in 179cd43a.

Joe4evr commented on docs/faq/Commands.md:
```diff
@@ -0,0 +1,142 @@
+return a successful [ExecuteResult] regardless of the execution.
+
+The following are the known caveats with `RunMode.Async`,
+	1. You introduce race condition.
```
Race conditions should be plural here. The phrase a a whole can also use the word "potential", but I don't have a good one atm.

Joe4evr commented on docs/faq/basic-operations.md:
```diff
@@ -0,0 +1,88 @@
+
+There are 2 ways to do this. You can do either of the following,
+	1. Cast the user as an [IGuildUser] and use its [IGuild] property.
+	2. Cast the channel as an [ITextChannel]/[IVoiceChannel] and use 
```
Should be "Cast as an `IGuildChannel`".

Joe4evr commented on docs/faq/basic-operations.md:
```diff
@@ -0,0 +1,88 @@
+
+Any implementation of [IMessageChannel] has a [SendMessageAsync]
+method. You can get the channel via [GetChannel] under the client.
+Remember, when using Discord.NET, polymorphism is a common recurring 
```
Should have a note that the returned channel object could also be an `IVoiceChannel`, which would be easier to understand that text messages can't be sent to such a channel.

Still34 replied:
Added in edb64bd.

Joe4evr commented on docs/faq/basic-operations.md:
```diff
@@ -0,0 +1,88 @@
+This is due to how .NET parses the HTML header, mistreating 
+0.25sec/action to 1sec. This casues the lib to throw preemptive rate 
+limit more frequently than it should for methods such as adding 
+reactions.
```
Is this really a .NET-specific bug? I thought it was an HTTP bug that header values can't be fractions, thus affecting *every* lib (and the others just get around it by implementing a special case).

Still34 replied:
Fixed in edb64bde.

Joe4evr commented on docs/faq/getting-started.md:
```diff
@@ -0,0 +1,73 @@
+First of all, welcome! Before you delve into using the library; 
+however, you should have some decent understanding of the language 
+you are about to use. This library touches on 
+[Task-based Asynchronous Pattern], [polymorphism], [interface] and 
```
Could add `(TAP)` to show off the abbreviation.

Joe4evr commented on docs/faq/getting-started.md:
```diff
@@ -0,0 +1,73 @@
+There are 2 ways to obtain the said ID. 
+  1. Enable Discord's developer mode. With developer mode enabled, 
+  you can - as an example - right click on a guild and copy the guild 
+  id (please note that this does not apply to Role IDs, see below).
```
This method also doesn't work for grabbing DM channel IDs in the client (you'll only get the target user's ID, which is *not the same ID*).

Joe4evr commented on docs/faq/samples/commands/Remainder.cs:
```diff
@@ -0,0 +1,19 @@
+[Command("echo-hassle")]
+public Task EchoAsync(string text) => ReplyAsync(text);
+
+//	The message would be seen as having 5 parameters, while the method 
```
*Five* parameters?

Joe4evr commented on docs/guides/commands/commands.md:
```diff
@@ -153,15 +153,15 @@ Invoke [CommandService.AddModulesAsync] to discover modules and
+[CommandService.AddModuleAsync]: xref:Discord.Commands.CommandService#Discord_Commands_CommandService_AddModuleAsync__1_System_IServiceProvider_
 
 ### Module Constructors
 
```
I can't add a comment below here, but `Alternatively, you may accept an
IServiceProvider as an argument and extract services yourself.` should be discouraged.

Still34 replied:
Added in edb64bd.

Still34 replied:
Added in edb64bd.

Joe4evr commented on src/Discord.Net.Commands/CommandError.cs:
```diff
@@ -1,26 +1,50 @@
+        /// </summary>
         ParseFailed,
+        /// <summary>
+        /// Thrown when the input text has too few parameters.
```
"too few *or too many*" and input is *arguments*, not parameters. :upside_down_face:

Joe4evr commented on src/Discord.Net.Commands/ModuleBase.cs:
```diff
@@ -21,11 +21,16 @@ public abstract class ModuleBase<T> : IModuleBase
-
+        /// <summary>
+        /// The method to execute before executing the command.
+        /// </summary>
```
Missing the `<param>`.

Commit:
Add permission calc & minor wording changes

Joe4evr reviewed (commented):
Things I noticed.

Commit:
Add changes according to comments from PR

Commit:
Add DI video link

Commit:
Fix broken markdown links

Commit:
Fix perm calc link

Commit:
Remove references to 1.0 & tidy links

Commit:
Add notification of potential difference in sample csproj

Commit:
Remove 1.0 ref

Commit:
Add warning against direct cast & fix broken MD link

Commit:
Fix broken indent

Commit:
Reword legacy versions

Commit:
Change FAQ structure to tree

Commit:
Remove overwrite from docfx.json

Commit:
Enable docs search

Commit:
Vastly simplified xref redirection

Commit:
Add XMLDocs for common entities

+ DiscordConfig
+ CDN
+ MentionUtils
+ LogMessage
+ LogSeverity

Commit:
Add XMLDocs

+ EmbedBuilderExtensions
+ CacheMode
+ ImageFormat

Commit:
Add XMLDocs

+ ActivityType
+ IAttachment
+ ChannelPermission
+ ChannelPermissions
+ ChannelType
+ Attachment

Commit:
Add XMLDocs

* Fix comments to comply with third-person commenting style
+ Various attributes
+ Various Command-related objects
+ Many more

Commit:
Add XMLDocs

+ GameAsset
+ GameTimestamps
+ Format

Commit:
Inital post-execution draft

Commit:
Add Nsfw precondition into bundle

Commit:
Warning-ify long running code part in commands

Commit:
Add Task<RuntimeResult> in command section

Commit:
Linkify Type Readers in Commands

Commit:
Fix CheckPermission in docs

Commit:
Add missing variable name in module example

Commit:
Improve readability & general fixes

Notable changes include the following,

+ Add _services into  examples.
+ Rename Commands back to commands.
    - This was changed in my previous PR; however, this made readability really weird.
+ Fix broken examples.
+ Directly referencing anchors instead of saying 'see this below'

Commit:
Improve readability for module example

Commit:
Fix ReadAsync for TypeReader example

Commit:
Fix xrefing for foreign objects

Commit:
Add missing ModuleBase<T> in DI example

Commit:
Add XMLDocs

To many files.

Commit:
Add extension comments

Commit:
Merge branch 'docs/faq-n-patches' of https://github.com/Still34/Discord.Net into docs/faq-n-patches

Commit:
Add 'View Source' button

+ This change requires us to target the solution.
+ Due to docfx issues, we must specify a  upon metadata compilation.
    - Because of this, docfx may complain about missing targets.
    - If we don't target a specific TargetFramework, docfx will fail to recognize documents with multiple framework targets.
    - Further investigation may be required.

Commit:
Target individual projects instead of solution

+ This gets rid of unnecessary compilation for the test project.

Commit:
Add workaround for merging framework targets

Commit:
Fix typo

Commit:
Rollback to target NS1.1 only

Commit:
Target NS1.3

Commit:
Add XMLDocs for some channel entities

Commit:
Merge dev changes

Still34 commented:
@foxbot what should we do about missing `TargetFramework` for `Discord.Net.Commands` and `Discord.Net.Webhook`? Ideally those should target NS1.3 as well for them to compile properly during docfx metadata compilation.

Commit:
Replace True keyword with langword xmldocs

Commit:
Add XMLDocs

+ This commit also adds overwrites for CommandContexts; this allows for additional remarks for the class.

Commit:
Add XMLDocs

Commit:
Fix DI module example

Commit:
Add Overwrites files for some classes

+	Starting from this commit, there will be overwrite files added to
	provide further details about certain APIs.

Commit:
Add XMLDocs

Commit:
Add XMLDocs

Commit:
Cleanup FAQ trailing spaces

Commit:
Add more embed example details

Commit:
Initial dark theme support

Commit:
Add dotnet CLI example

Commit:
Update dark theme

+ Fix table color.
+ Change warning/tip/important tip color.
+ Change font to Titillium Web.
+ Other changes that's easier on eyes.

Commit:
Improve Overwrites styling

Commit:
Add lightbox & categorize templates

Commit:
Clean up pages

Commit:
Further styling improvement

Commit:
Add Xmldocs

Commit:
Tidy up docs

Commit:
Add PreconditionAttribute docs

Commit:
Add more examples

Commit:
Merge branch 'docs/faq-n-patches' of https://github.com/Still34/Discord.Net into docs/faq-n-patches

Commit:
Merge with 109f663a9adc450823f9cd91da03812256553f51

Commit:
Target .NS1.3 for Commands and Webhook

Commit:
Increase font-size to 15px

Commit:
Improve landing page & add intro

Commit:
Improve readability

Commit:
Add 'preemptive rate limit' section

Commit:
Add titles and UIDs to each article

Commit:
Add examples for precondition attributes

Commit:
Simplify CSS

+ Changes dark theme changes to a different css to respect future docfx.css changes
+ Merge featherlight plugin into the dark theme

Commit:
Prep for theme switcher

Commit:
Implement rudimentary support for theme switching

Commit:
Improve theme selection

+ Stylized option menu
+ The menu now defaults to whatever theme being the most preferred one

Commit:
Move theme-switcher next to sideaffix & fix contribution link hover

Commit:
Move from botched VS-like CSS to VS2015 CSS from HLJS repo

Commit:
Fix file readability

Commit:
Fix minor markdown issues

Commit:
Fix headers in command guide & add mention of nullable types

Commit:
Add warning regarding TR addition prior to module

Commit:
Draft for complete post-execution guide

Commit:
Improve readability of command guide

Commit:
Fix xref in post-execution

Commit:
Change all .NET to .Net

This change is because how our repo and project namespace uses .Net instead of .NET.
For consistency sake, we should refer to the project as Discord.Net.

Commit:
Finish structuring of post-execution guide

Commit:
Add docs for async enumerable extensionn

Commit:
Fix minor wordings

Commit:
Add RuntimeResult demo

Commit:
Improve wording of post-execution

Commit:
Simplify RuntimeResult section

Commit:
Improve readability

Commit:
Remove redundant wording

Commit:
Cleanup docfx.json

Commit:
Fix casing of 'discord'

Commit:
Add Codealike to .gitignore

+ Codealike is a Visual Studio extension for developers to track their time spent on projects.
  It may drop additional file called codealike.json at the project root.

Commit:
Improve styling

+ Increase line-height to make pages look less squished.
+ Add master stylesheet.
	- Move content that are not theme-specific to master CSS.
+ Target latest featherlight.js instead of 1.7.6.
+ Add javascript for cornerifying images.
+ Add escape mention image, courtesy of @Andrewww1.

Commit:
Revert to targeting specific versions for JS

Commit:
Fix ToC view on mobile

Commit:
Merge branch 'docs/faq-n-patches' of https://github.com/Still34/Discord.Net into docs/faq-n-patches

Commit:
Improve readability for command service section

Commit:
Fix broken MD linking

Commit:
Split command service guide into various articles

+ Each section has gone through a thorough proof-reading and cleanup
+ This should help with readability.

Commit:
Improve various articles

+ Rename Guides.Concepts.Logging title from 'Logging' to 'Logging Events/Data'.
+ Rewrite various sections of Guides.GettingStarted.FirstBot and Guides.GettingStarted.Installing based on suggested changes from #1027.
+ Rename project.csproj example to project.xml to avoid auto-compilation from Visual Studio Code.
+ Rename 'samples\intro' to 'samples\first-bot' to reflect on the article name change.
+ Clean first-bot samples.

Commit:
Raise warning level for exception handling in first-bit.md

Commit:
Fix broken xref in first-bot

Commit:
Fix minor typos

Commit:
Add xmldocs

Commit:
Merge remote origin

Commit:
Merge remote-tracking branch 'origin/docs/faq-n-patches' into docs/faq-n-patches

Commit:
Add description metatag

Commit:
Revert description meta tag

Commit:
Cleanup emoji docs

Commit:
General cleanup

Commit:
Add ConfigureAwait(false) for most missing tasks

Commit:
Add Mono warning

Commit:
Inital draft for deployment

Commit:
Add a whole bunch of XML docs

Commit:
Add XML docs

Commit:
Add xml docs

Commit:
Add title to docs

Commit:
Merge branch 'docs/faq-n-patches' of https://github.com/Still34/Discord.Net into docs/faq-n-patches

Joe4evr commented on src/Discord.Net.Commands/Attributes/PreconditionAttribute.cs:
```diff
@@ -1,16 +1,22 @@
+        ///     <para>
+        ///          <see cref="Preconditions" /> of the same group require only one of the preconditions to pass in
+        ///          order to be successful (A || B). Specifying <see cref="Group" /> = <see langword="null" /> or not
+        ///          at all will require *all* preconditions to pass, just like normal (A && B).
```
Revert that `&&` back to `&amp;&amp;`.

Commit:
Add XML docs

foxbot committed:
Clean up style switcher

Commit:
Merge remote-tracking branch 'origin/docs/faq-n-patches' into docs/faq-n-patches

foxbot committed:
Squash commits on branch docs/faq-n-patches

Commit:
Resolve conflict

Commit:
Fix broken theme selector

Commit:
Merge dev

Commit:
Add local image embed instruction

Commit:
Add a bunch of XML docs

Commit:
Add a bunch of XML docs

Commit:
Fix broken search
+ DocFX by default ships with an older version of jQuery, switching to a newer version confuses parts of the DocFX Javascript.

Commit:
Minor fixes for CONTRIBUTING.md and README.md

Commit:
Clean up filterConfig.yml

+ New config exposes Discord.Net namespace since it has several common public exceptions that may be helpful to users

Commit:
Add XML docs

Commit:
Merge upstream/dev & resolve conflicts

Commit:
Read token from Environment Variable instead of hardcode

Commit:
Add XMLDocs

Commit:
Squashed commit of the following:

commit dece19d89505342e5f8c51317b54be8ce69336c9
Merge: fdaa689a e764dafe
Author: Still Hsu <341464@gmail.com>
Date:   Mon May 21 11:47:47 2018 +0800

    Merge branch 'dev' into docs/faq-n-patches-offline

commit e764dafe083bf5db62091ac1f6b3d438eee38882
Author: Quahu <quahuu@gmail.com>
Date:   Sun May 13 15:34:40 2018 +0200

    Add ViewChannel to Voice channel permissions (#1059)

    Previously, Voice channels did not have ViewChannel in their "all" permissions

commit 32fc2df21b1840fd913ccce80c0686e8e853b43d
Author: Alex Gravely <tcbskater@hotmail.com>
Date:   Sat May 12 20:47:44 2018 -0400

    Remove unused field in EmbedFieldBuilder. (#1018)

commit 39dffe858584d0e9eed750bff6426e9562db4262
Author: Finite Reality <FiniteReality@users.noreply.github.com>
Date:   Sun May 13 01:46:07 2018 +0100

    Audit Logs implementation (#1055)

    * Copy audit logs impl from old branch and clean up

    I suck at using git, so I'm gonna use brute force.

    * Remove unnecessary TODOs

    Category channels do not provide any new information, and the other
    I forgot to remove beforehand

    * Add invite update data, clean up after feedback

    * Remove TODOs, add WebhookType enum for future use

    WebhookType is a future-use type, as currently audit logs are the only
    thing which may return it.

commit fdaa689ae8be8e9d2ff5190b6cfed5af244b805b
Author: Still Hsu <341464@gmail.com>
Date:   Wed May 9 06:04:59 2018 +0800

    Add explanation for RunMode

commit ea82c2537e64d4b75f85d9369e90cb55cfcadcc7
Author: Still Hsu <341464@gmail.com>
Date:   Tue May 8 16:30:48 2018 +0800

    Initial proofread of the articles

commit 124f1a267c663fbbb0f2dc23367dd910ae3ba4cc
Merge: 25557218 97c89310
Author: Still Hsu <341464@gmail.com>
Date:   Tue May 8 05:02:01 2018 +0800

    Merge branch 'dev' into docs/faq-n-patches-offline

commit 97c893107b51de24a38ea03c2f8260030c8fc7f5
Author: Still Hsu <341464@gmail.com>
Date:   Mon May 7 06:22:49 2018 +0800

    Implement GetBanAsync (#1056)

commit 25557218dbc9390dbeac108afc153776f52dc26f
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 16:11:19 2018 +0800

    Add details to SpotifyGame

commit c7b236ddf5d7324dd8c1513ccd528a7a45d0031f
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 15:58:23 2018 +0800

    Add more IGuild docs

commit 1bb06cc37b7c2312b0ab111da63026a01da01e42
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 15:40:31 2018 +0800

    Replace all langword placements with code block

commit ac47d84ea74c81e136434dcc33967fe4cd95c1e6
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 15:22:17 2018 +0800

    Replace langword null to code block null instead

    - Because DocFX sucks at rendering langword

commit 0b15bbc54d15c036f770bab818712956c527b4ee
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 15:20:34 2018 +0800

    Add XML docs

commit 65d4e4360eb88e3bc5cc8f0da6135e5ee997af6c
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 06:57:53 2018 +0800

    Add BaseSocketClient docs

commit 8f64c045999c1bb0440bdc818f082e7459756b36
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 06:37:55 2018 +0800

    Replace note block

commit d8bb9e7aaa1e908ceb9227285c0cebec9add5280
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 06:31:50 2018 +0800

    Add warning for bulk-delete endpoint

commit adae5ffc9e42343f170d6c869644637587330b59
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 06:07:28 2018 +0800

    Fix missing Username prop

commit 3e591972cafdffb6692a1b94cf9a63251188df8b
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 06:01:34 2018 +0800

    Add properties examples to overwrite

commit 0ad66f6765addf6d4d25dfbddadeaa542d461403
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 04:55:15 2018 +0800

    Fix minor consistencies & redundant impl

commit 124efdf7e66ee070fbcdde07ed4f20c717c9c1c0
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 04:40:14 2018 +0800

    XML Docs

commit 3aa5d363de335444558fdb4bf1bf48a5743bc917
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 18:22:46 2018 +0800

    Add 'last modified' plugin

    Source: https://github.com/Still34/DocFx.Plugin.LastModified
    Licensed under MIT License

commit 2014870dc05dedb6171338d8ae4329eaa4c73f90
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 15:57:40 2018 +0800

    Fix letter-casing for files

commit f27d659ebeb88348e24e759308933e3983831ad7
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 15:50:00 2018 +0800

    Document exposed TypeReaders

commit 5a824a5695d72a62e28e0a307a5be446557291be
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 15:44:15 2018 +0800

    Add missing exceptions

commit c2de0c055f3894c795ea7d7da774fb006c752898
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 15:40:16 2018 +0800

    Fix seealso for preconditions and add missing descriptions

commit 3a7d7ee9552a2c6fea1c32b6f3cd91f6efdd29f5
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 15:36:22 2018 +0800

    Minor fixes in documentations
    + Fix unescaped '<'
    + Fix typo

commit 45839bd982a822dd9da10ceb48989fdb2cd8cd85
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 15:29:47 2018 +0800

    Add XML Docs

commit 9e6254600cd7bb5b7626025492223717f92501f9
Merge: aea06788 bb4bb138
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 14:48:13 2018 +0800

    Merge branch 'dev' into docs/faq-n-patches-offline

commit aea067884c81832ad6cd910ff56c550cf3ecad75
Merge: 27dc4831 9ddd7090
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 13:51:07 2018 +0800

    Merge branch 'docs/faq-n-patches-offline' of https://github.com/Still34/Discord.Net into docs/faq-n-patches-offline

commit 27dc4831e88c952f3ea1f68b18434e16197fdfa9
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 13:50:27 2018 +0800

    Add XML Docs

commit baa8beb3824ed59fb50d269b9e13721fa9f5a04e
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 13:49:42 2018 +0800

    Add XML Docs

commit 089f97a0101a4ef9194f65b2d4435c0c69dfa87c
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 13:43:04 2018 +0800

    Add details regarding userbot support

commit bb4bb138460cc1e6bac53b4e838893dcfd812ce7
Author: Finite Reality <FiniteReality@users.noreply.github.com>
Date:   Fri May 4 11:42:54 2018 +0100

    Fix issues with #984, remove extraneous whitespace (#1051)

    - Removed unnecessary parameter in SocketVoiceServer

    - Moved SocketVoiceServer into Entities/Voice

    - Fixed a bug where trying to download the cached guild would throw

    - Fixed a potential bug where Discord might not give us a port when
      connecting to voice

commit 9ddd70906aa01015fc76bc2401868e044c4c7cc1
Merge: f197174f 7cfed7ff
Author: Still Hsu <341464@gmail.com>
Date:   Fri May 4 09:37:04 2018 +0800

    Merge branch 'dev' into docs/faq-n-patches-offline

commit f197174fcc2df1e092b26cd2c8087b602ffb1960
Author: Still Hsu <341464@gmail.com>
Date:   Fri May 4 09:36:53 2018 +0800

    Fix embed docs consistency

commit 7cfed7ff67ac9aee517de4130d9ccf504791ed61
Author: Alex Gravely <tcbskater@hotmail.com>
Date:   Thu May 3 21:30:13 2018 -0400

    Fix nullref when passing null to GetShardIdFor. (#1049)

commit e775853b1b26eaf80e6e76c295b44ca6241eaee7
Author: Luke <dev@crimsonxv.pro>
Date:   Fri May 4 02:29:51 2018 +0100

    Expose VoiceServerUpdate events (#984)

    * Expose VoiceServerUpdate events

    * Amend based on feedback

    * Move this out of guild entity

    * Fix namespace issue

    * Adjust based on feedback #2

    * Use cacheable instead

    * Change based on feedback

commit 157acc46955d52809b959b52fb1a04404a6df3a5
Author: Still Hsu <341464@gmail.com>
Date:   Thu May 3 23:03:35 2018 +0800

    Add tag examples

commit 57ea571a8115f4b768c2dbbe1f2ef807b0cfd647
Author: Still Hsu <341464@gmail.com>
Date:   Thu May 3 22:48:33 2018 +0800

    Fix sample link & add missing pictures

commit 6769c37ad7e664845948c50ce403217a9951ef07
Author: Still Hsu <341464@gmail.com>
Date:   Thu May 3 22:39:26 2018 +0800

    Compress some assets & add OAuth2 URL generator

Commit:
Merge branch 'dev' into docs/faq-n-patches

Still34 commented:
Just FYI, this is currently being worked on on a separate branch `docs/faq-n-patches-offline`, and it is will be squash merged occasionally to prevent spam on the PR branch.

Commit:
Squashed commit of the following:

commit 17a4af71ee8062966bfe477d4d9ea68b490cfb77
Author: Still Hsu <341464@gmail.com>
Date:   Tue May 29 17:15:10 2018 +0800

    Add documentation for new API version & few events

commit f4fb027ea9cab9ba990a55676d8453bea3e68b1a
Merge: d74c5287 3acf2a9a
Author: Still Hsu <341464@gmail.com>
Date:   Tue May 29 08:25:41 2018 +0800

    Merge remote-tracking branch 'upstream/dev' into docs/faq-n-patches-offline

commit 3acf2a9a6bb13d5b75c09babaf7234cb2707f4e8
Author: HelpfulStranger999 <helpfulstranger999@gmail.com>
Date:   Mon May 28 14:59:30 2018 -0500

    Refactors readMessages, fixing from #1033 (#1075)

commit 513a489bf82ed4e845ff3df9fe21c06e80bb28b8
Author: Christopher F <foxbot@protonmail.com>
Date:   Mon May 28 15:45:42 2018 -0400

    codefix: add VoiceAPIVersion constant to DiscordConfig

commit 9ba38d77966897037cb6355eda2215fd9a235712
Author: Christopher F <foxbot@protonmail.com>
Date:   Mon May 28 15:35:59 2018 -0400

    api: upgrade audio client to VoiceWS v3

commit 64d8938ed5100d4c8f3a2afaec3a6becb53cf9b8
Author: Christopher F <foxbot@protonmail.com>
Date:   Mon May 28 15:25:46 2018 -0400

    fix: Discord does not always send the member on VOICE_STATE_UPDATE

    some guilds aren't on g250k yet? not sure

commit aeb30955932803e88fcd056f817f449e8664715f
Author: FiniteReality <digpoealternate@gmail.com>
Date:   Mon May 28 17:48:38 2018 +0100

    Remove Debug.Print call causing tests to fail

    Forgot to hit save properly... Whoops.

commit a718a7d3c234a37d1c792ac84af0ae8dce61c1a5
Author: Christopher F <foxbot@protonmail.com>
Date:   Mon May 28 12:45:49 2018 -0400

    cleanup: Remove ChannelHelper#IsNsfw

    Closes #1074

    This code was not referenced anywhere in the active codebase.

commit 0c7cb73b06aba3eaeb811b4671fd0f4a5078a491
Author: FiniteReality <digpoealternate@gmail.com>
Date:   Mon May 28 17:41:05 2018 +0100

    Fix permission-related ViewChannel tests

commit 415e2f773978353025ea8b2f6c3d9929601c28c1
Author: FiniteReality <digpoealternate@gmail.com>
Date:   Mon May 28 17:14:18 2018 +0100

    Fix analyzer warnings with tests

    Use (actual, expected) format for Assert.Equal, fixes analyzer warnings

commit b0042606b628e02ffe2bee3135c7ed42c3b8311f
Author: FiniteReality <digpoealternate@gmail.com>
Date:   Mon May 28 17:13:59 2018 +0100

    Move RpcCategoryChannel into the correct project

commit 683b3f2701087db6d49c3a54c6d99c80ef5b2e5a
Author: FiniteReality <digpoealternate@gmail.com>
Date:   Mon May 28 17:12:28 2018 +0100

    Mark ChannelHelper.IsNsfw Obsolete

    Discord no longer treats channels prefixed with 'nsfw' as NSFW,
    so we no longer need to have this check.

commit d74c52873fb13434bca9dd8b56e74104392cb95d
Merge: 341b9588 237ad0f8
Author: Still Hsu <341464@gmail.com>
Date:   Mon May 28 11:48:47 2018 +0800

    Merge branch 'dev' into docs/faq-n-patches-offline

commit 237ad0f8675023478ec8ad41c7b6b6fd19ceed2d
Author: Christopher F <foxbot@protonmail.com>
Date:   Sun May 27 18:00:51 2018 -0400

    Revert "fix: Properly rethrow exceptions in SocketGuild audio client"

    This reverts commit adf4da19fc116759c4246c978143987712fa2116.

    Someone hasn't written c# in a while

commit adf4da19fc116759c4246c978143987712fa2116
Author: Christopher F <foxbot@protonmail.com>
Date:   Sun May 27 17:56:14 2018 -0400

    fix: Properly rethrow exceptions in SocketGuild audio client

commit c0c565fd7ee03fa569027fff7f17db90d9376b4f
Author: Christopher F <foxbot@protonmail.com>
Date:   Sun May 27 16:45:54 2018 -0400

    retain fallback case for if user is still null

commit 8fb2c71814fad9bcab1888fb8d66d693cc98a4b1
Author: Christopher F <foxbot@protonmail.com>
Date:   Sun May 27 16:37:17 2018 -0400

    Add new member objects to events

commit 341b9588d608bcd7460e751b7b0d2cd83bc42e32
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 27 19:01:04 2018 +0800

    Add documentation

commit 2df635e025ef7de2959c232e468f093c103f250f
Merge: 180b20dd 9995cab4
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 27 18:25:51 2018 +0800

    Merge pull request #1 from Chris-Johnston/better-documentation-for-quotation-mark-aliases

    Improve XML docs for commands quotation mark aliases feature

commit 180b20dd02edddfe1d279348516f5d54985eb8f0
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 27 18:19:20 2018 +0800

    Add documentation for new INestedChannel

commit e9f72b6ece60ffe142266b01f90bd5ae8b00c742
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 27 18:18:52 2018 +0800

    Fix missing service provider in example

commit 9995cab40ce542562d87183cff91707bd6f07e60
Merge: a8d3031b 1a146afb
Author: Chris Johnston <chris@thejohnstons.net>
Date:   Sun May 27 01:11:17 2018 -0700

    Merge branch 'docs/faq-n-patches-offline' into better-documentation-for-quotation-mark-aliases

commit a8d3031b2d73bbe813e83bb439345975d1003b55
Author: Chris Johnston <chris@thejohnstons.net>
Date:   Sat May 26 23:32:40 2018 -0700

    Add explanation that if " is removed then it wont work

commit cb7c6d35464bd2d8aa0d5cc0dd20c8a6fa11bb4f
Author: Chris Johnston <chris@thejohnstons.net>
Date:   Sat May 26 23:29:32 2018 -0700

    Add reference to CommandServiceConfig from the util docs'

commit 110b97513fbd26695674190ab217800490e27185
Author: Chris Johnston <chris@thejohnstons.net>
Date:   Sat May 26 23:15:01 2018 -0700

    Add more XML comments to quotation mark alias map stuff, including an example

commit 1a146afbe1f539c210648c39985009bd4ad926b3
Merge: cb57ada8 a06e2126
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 27 11:25:10 2018 +0800

    Merge branch 'dev' into docs/faq-n-patches-offline

commit a06e21261c66e7aeacde11911ce5ee1ed6baba2f
Author: Chris Johnston <chris@thejohnstons.net>
Date:   Sat May 26 11:15:09 2018 -0700

    Fix ChannelPermissions Modify parameter to be correct default value (#1003)

    * fix channel permissions modify parameter to use nullable boolean, correct default value

    * Add general tests for the ChannelPermissions.Modify method to test default values

    * remove unused cast in tests

    * add guildpermission modify no param tests

    * Add no-param modify tests for OverwritePermissions

    * fix inconsistent parameters in GuildPermissions cstr

    * Adjust formatting of methods and cstrs with many parameters

    * remove temp file that was included. no idea what that is

    * Fix System dependency

    I should really stop fixing merge conflicts in the github website.

commit f9cbff5e42c04aa8a0fe87377ba68aa7d748d873
Author: Chris Johnston <chris@thejohnstons.net>
Date:   Sat May 26 11:06:35 2018 -0700

    Fix #995, Move Category Implementation from IGuildChannel to INestedChannel (#1004)

    * Fix #995 ICategoryChannel.CategoryID throws NotSupportedException

    * Add tests

    * change run mode of TestChannelCategories

    * Add throw for GetCategoryAsync

    * Add xml doc explaining why exception is thrown

    * Add test coverage for text and voice channel categories

    * initial implementation of INestedChannel

    * more implementation of INestedChannel design

    * Add case in RestChannel Create for Category type

    * set the CategoryID for RestVoiceChannel

    * rewrite channel category tests to work with existing pattern

    * remove outdated todo

    * Make IVoiceChannel implement INestedChannel

    * remove redundant interface implementation

    * Add c#7 feature from feedback

    * Remove redundant GetCategoryAsync methods from socket entities

    * Added configureawait to async methods

    * change signature of interface GetCategoryAsync

    * Add check for cachemode in rest channel GetCategory

    * remove redundant IGuildChannel interface from ITextChannel and IVoiceChannel

commit 4d8764e124cef12af4b0944d185ec22d9683c520
Author: FiniteReality <digpoealternate@gmail.com>
Date:   Sat May 26 14:11:49 2018 +0100

    Refactor RoleInfo types into separate classes

    - Fixes indentation with a few files (looks like a bad copy+paste job)
    - Renames RoleInfo to RoleEditInfo
    - Moves RoleInfo from MemberRoleAuditLogData into its own class

    All of this should improve docs.

commit cb57ada8d9f4a8a8b787ba6990afb08ea81a89a1
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 26 21:03:07 2018 +0800

    Add partial documentation for audit log objects

commit afda0cd172a98154bc33b4c27eed65580189785d
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 26 20:47:21 2018 +0800

    Add documentation for some REST-based objects

commit 098ead93d98ea9e629f9c4d8681a38ad8c1e9a82
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 26 16:33:38 2018 +0800

    Add partial documentation for audit log impl

commit 5d1621a9c5ea27d8758e6a6e6dc4b4d6c093ff7a
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 26 16:06:20 2018 +0800

    Fix xml docs error

commit 596b47e1b8bbdd349df20da7938eaf4581bccb91
Merge: 3f572ccc fa759a22
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 26 15:54:08 2018 +0800

    Merge branch 'dev' into docs/faq-n-patches-offline

commit 3f572ccc0d10fbdcc514f5ef0701b06c8b75e7ec
Merge: 73ebc025 a9983026
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 26 15:32:38 2018 +0800

    Merge branch 'docs/faq-n-patches-offline' of https://github.com/Still34/Discord.Net into docs/faq-n-patches-offline

commit 73ebc0258031ff74770b1f4b4a8c4be9071f703c
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 26 15:32:23 2018 +0800

    Fix light theme link color

commit fa759a22eac25183eb0fecf6cd0fbbf7f476fa11
Author: Chris Johnston <chris@thejohnstons.net>
Date:   Fri May 25 06:38:59 2018 -0700

    Remove Build Warnings for Obsolete Guild Permission Read Messages (#1067)

    - Replaces the usages of `ReadMessages` with `ViewChannel`
    - Renames the read message parameters of `GuildPermissions#Modify` to be view channel as well

commit bbbac85c46162d7bb44ecdd68194ab79db25037d
Author: Hawx <Hawxy@users.noreply.github.com>
Date:   Fri May 25 20:14:28 2018 +0800

    Update Dependencies and support NS2.0 builds (#1046)

    * Update deps, fix test warnings.

    * Support ns2.0

    * Fix typo

    * Remove ns1.1 support

    * Net.Http and Net.Websockets.Client are not needed in ns2.0

    * Move to net46 per volt

    * Remove ns1.3 constants

commit 5f084adf94f3215b6d68c3a8b61cbfad7376f18c
Author: HelpfulStranger999 <helpfulstranger999@gmail.com>
Date:   Thu May 24 19:17:19 2018 -0500

    Deprecates ReadMessages, introduces ViewChannel (#1033)

    * Deprecates ReadMessages, introduces ViewChannel

    * Adds period and comma somehow missed

commit bc6009ec7207e16aedecdc658dc59b8ac02c400e
Author: Still Hsu <341464@gmail.com>
Date:   Fri May 25 08:08:51 2018 +0800

    Implement IMessageChannel#DeleteMessageAsync (#996)

    * Implement DeleteMessageAsync

    * Refer to MessageHelper instead of duplicating call

    * Fix refactor error

commit cee71ef35a450802ef35090d9eaf167ca39306e8
Author: Chris Johnston <chris@thejohnstons.net>
Date:   Thu May 24 17:07:37 2018 -0700

    Add support for parsing multiple types of quotation marks in commands, Fix #942 (#943)

    * Add ability to support different types of quotation marks

    * Added normal quotation mark to list of aliases, removed single quote mark

    * clean up leftover changes from testing

    * change quotation mark parsing to use a map of matching pairs

    * remove commented out code

    * Fix conventions of the command parser utility functions

    * change storage type of alias dictionary to be IReadOnlyDictionary

    * revert type of CommandServiceConfig QuotationMarkAliasMap to Dictionary

    * minor formatting changes to CommandParser

    * remove unnecessary whitespace

    * Move aliases outside of CommandInfo class

    * copy IReadOnlyDictionary to ImmutableDictionary

    * minor syntax changes in CommandServiceConfig

    * add newline before namespace for consistency

    * newline formatting tweak

    * simplification of GetMatch method for CommandParser

    * add more quote unicode punctuation pairs

    * add check for null value when building ImmutableDictionary

    * Move default alias map into a separate source file

    * Ensure that the collection passed into command service is not null

commit b52af7ae7caaddd79101aeb9f5af2ce482890bc2
Author: Joe4evr <jii.geugten@gmail.com>
Date:   Fri May 25 01:59:32 2018 +0200

    Add a dedicated TimeSpan reader so it doesn't suck (#1005)

    * Add a dedicated TimeSpan reader so it doesn't suck

    * Pass input as lower case

commit c275e575289073fe42ead725de97b97301c7dc50
Author: o Acid Chicken (硫硫▒~E▒▒▒~O) <root@acid-chicken.com>
Date:   Thu May 24 19:52:25 2018 -0400

    Add support casting System.Drawing.Color to Discord.Color

    commit fa3303426766a59e7aa4d67e8b50826cfe7204ec
    Author: Acid Chicken (硫酸鶏) <root@acid-chicken.com>
    Date:   Sun Apr 29 13:50:56 2018 +0900

        Use built-in symbol

        refs: https://github.com/RogueException/Discord.Net/pull/1043#issuecomment-385223999

    commit 27ea82668587960925b73f97c924c9d39ee71f7b
    Author: Acid Chicken (硫酸鶏) <root@acid-chicken.com>
    Date:   Sun Apr 29 11:29:31 2018 +0900

        Add support casting System.Drawing.Color to Discord.Color

    commit 1ab9de24978ff24a018767f80dc95ba19b616988
    Merge: f5bb99c7 a4d1e2bc
    Author: Acid Chicken (硫酸鶏) <root@acid-chicken.com>
    Date:   Sun Apr 29 10:16:46 2018 +0900

        Merge remote-tracking branch 'upstream/dev' into dev

    commit f5bb99c77d0fecec21ad769778d17144c047d8b1
    Merge: 3be8e40d b8b59d97
    Author: Acid Chicken (硫酸鶏) <root@acid-chicken.com>
    Date:   Tue Apr 10 11:07:36 2018 +0900

        Merge remote-tracking branch 'upstream/dev' into dev

    commit 3be8e40d3814fe8455af475b8641ee07a8f368d6
    Merge: c692306f 9d77a3cd
    Author: Acid Chicken (硫酸鶏) <root@acid-chicken.com>
    Date:   Sun Jan 7 15:05:31 2018 +0900

        Merge remote-tracking branch 'upstream/dev' into dev

    commit c692306fcc1e86ab92dd10683d3719f16c02a249
    Author: Acid Chicken (硫酸鶏) <root@acid-chicken.com>
    Date:   Sat Nov 11 10:49:00 2017 +0900

        Add target of the internal fields

    commit 2d08f9a655b4949c1177f778d0f499047484a537
    Author: Acid Chicken (硫酸鶏) <root@acid-chicken.com>
    Date:   Sat Nov 11 10:17:42 2017 +0900

        Add some more extension

    commit 4f19b835ffe8c64a93a9b4659e60b03ac797760f
    Author: Acid Chicken (硫酸鶏) <root@acid-chicken.com>
    Date:   Sat Nov 11 01:19:11 2017 +0900

        Add naming rules

    commit af756cd9feb630baadbf6025cbb079cd9e1f45cb
    Author: Acid Chicken (硫酸鶏) <root@acid-chicken.com>
    Date:   Sat Nov 11 00:35:30 2017 +0900

        Add basic .NET style rules

    commit 503ece558b4f07bd8008157d3aeb6a4e7100d349
    Author: Acid Chicken (硫酸鶏) <root@acid-chicken.com>
    Date:   Fri Nov 10 22:36:52 2017 +0900

        Add EditorConfig

commit 5023357a60bd5ae454b35426ddd279fb3284b7e8
Author: Christopher F <foxbot@protonmail.com>
Date:   Thu May 24 19:36:33 2018 -0400

    codefix #1020: RequestOptions should always be sorted last

commit bf5275e071b78a25fd4501a930e69c5dc8cd3b19
Author: Joe4evr <jii.geugten@gmail.com>
Date:   Thu May 24 19:36:00 2018 -0400

    Add ability to specify parameters on channel creation (#1020)

    commit 07bca5b31a3580d55278878eabb56a82973f8c8f
    Author: Joe4evr <jii.geugten@gmail.com>
    Date:   Fri Apr 6 09:44:50 2018 +0200

        Add ability to specify parameters on channel creation

commit 0ba8b063ad98144232606f422574f9741fd95b7a
Author: HelpfulStranger999 <helpfulstranger999@gmail.com>
Date:   Thu May 24 19:23:44 2018 -0400

    Makes text parameter of sending messages optional (#1042)

    commit 114e5b431b26669bcdaac9f84792a216ad67186f
    Author: HelpfulStranger999 <helpfulstranger999@gmail.com>
    Date:   Sat Apr 28 19:08:35 2018 -0500

        Fixes lack of default value for tts

    commit 1fd8c70c5346ff0c4fdb0093c0fc7fb4b3c8db2c
    Author: HelpfulStranger999 <helpfulstranger999@gmail.com>
    Date:   Sat Apr 28 15:21:11 2018 -0500

        Makes text parameter of sending messages optional

commit 79811d0e618c011e6cca542811031d824650dae5
Author: HelpfulStranger999 <helpfulstranger999@gmail.com>
Date:   Thu May 24 17:35:38 2018 -0500

    Paginate reactions - solved #1007 (#1022)

    * Cleaned up and refactored slightly

    * Resolves #971

    * Adds support for default avatars and resolves #971

    * Amendment

    * Final amendment

    * Paginating reactions

    * Amendments based on feedback

    * Further amendment based on review

    * Final(?) amendment

    * Removes default limit and after user id

    * Removes fromUserId; cleans up model creation; replaces function with individual parameters

commit a9983026f067effeac7108c710ba6917162b7995
Author: Still Hsu <341464@gmail.com>
Date:   Tue May 22 10:15:49 2018 +0800

    Add event docs

    - MessageReceived
    - ChannelUpdated/Destroyed/Created

commit dece19d89505342e5f8c51317b54be8ce69336c9
Merge: fdaa689a e764dafe
Author: Still Hsu <341464@gmail.com>
Date:   Mon May 21 11:47:47 2018 +0800

    Merge branch 'dev' into docs/faq-n-patches-offline

commit fdaa689ae8be8e9d2ff5190b6cfed5af244b805b
Author: Still Hsu <341464@gmail.com>
Date:   Wed May 9 06:04:59 2018 +0800

    Add explanation for RunMode

commit ea82c2537e64d4b75f85d9369e90cb55cfcadcc7
Author: Still Hsu <341464@gmail.com>
Date:   Tue May 8 16:30:48 2018 +0800

    Initial proofread of the articles

commit 124f1a267c663fbbb0f2dc23367dd910ae3ba4cc
Merge: 25557218 97c89310
Author: Still Hsu <341464@gmail.com>
Date:   Tue May 8 05:02:01 2018 +0800

    Merge branch 'dev' into docs/faq-n-patches-offline

commit 25557218dbc9390dbeac108afc153776f52dc26f
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 16:11:19 2018 +0800

    Add details to SpotifyGame

commit c7b236ddf5d7324dd8c1513ccd528a7a45d0031f
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 15:58:23 2018 +0800

    Add more IGuild docs

commit 1bb06cc37b7c2312b0ab111da63026a01da01e42
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 15:40:31 2018 +0800

    Replace all langword placements with code block

commit ac47d84ea74c81e136434dcc33967fe4cd95c1e6
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 15:22:17 2018 +0800

    Replace langword null to code block null instead

    - Because DocFX sucks at rendering langword

commit 0b15bbc54d15c036f770bab818712956c527b4ee
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 15:20:34 2018 +0800

    Add XML docs

commit 65d4e4360eb88e3bc5cc8f0da6135e5ee997af6c
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 06:57:53 2018 +0800

    Add BaseSocketClient docs

commit 8f64c045999c1bb0440bdc818f082e7459756b36
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 06:37:55 2018 +0800

    Replace note block

commit d8bb9e7aaa1e908ceb9227285c0cebec9add5280
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 06:31:50 2018 +0800

    Add warning for bulk-delete endpoint

commit adae5ffc9e42343f170d6c869644637587330b59
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 06:07:28 2018 +0800

    Fix missing Username prop

commit 3e591972cafdffb6692a1b94cf9a63251188df8b
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 06:01:34 2018 +0800

    Add properties examples to overwrite

commit 0ad66f6765addf6d4d25dfbddadeaa542d461403
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 04:55:15 2018 +0800

    Fix minor consistencies & redundant impl

commit 124efdf7e66ee070fbcdde07ed4f20c717c9c1c0
Author: Still Hsu <341464@gmail.com>
Date:   Sun May 6 04:40:14 2018 +0800

    XML Docs

commit 3aa5d363de335444558fdb4bf1bf48a5743bc917
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 18:22:46 2018 +0800

    Add 'last modified' plugin

    Source: https://github.com/Still34/DocFx.Plugin.LastModified
    Licensed under MIT License

commit 2014870dc05dedb6171338d8ae4329eaa4c73f90
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 15:57:40 2018 +0800

    Fix letter-casing for files

commit f27d659ebeb88348e24e759308933e3983831ad7
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 15:50:00 2018 +0800

    Document exposed TypeReaders

commit 5a824a5695d72a62e28e0a307a5be446557291be
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 15:44:15 2018 +0800

    Add missing exceptions

commit c2de0c055f3894c795ea7d7da774fb006c752898
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 15:40:16 2018 +0800

    Fix seealso for preconditions and add missing descriptions

commit 3a7d7ee9552a2c6fea1c32b6f3cd91f6efdd29f5
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 15:36:22 2018 +0800

    Minor fixes in documentations
    + Fix unescaped '<'
    + Fix typo

commit 45839bd982a822dd9da10ceb48989fdb2cd8cd85
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 15:29:47 2018 +0800

    Add XML Docs

commit 9e6254600cd7bb5b7626025492223717f92501f9
Merge: aea06788 bb4bb138
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 14:48:13 2018 +0800

    Merge branch 'dev' into docs/faq-n-patches-offline

commit aea067884c81832ad6cd910ff56c550cf3ecad75
Merge: 27dc4831 9ddd7090
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 13:51:07 2018 +0800

    Merge branch 'docs/faq-n-patches-offline' of https://github.com/Still34/Discord.Net into docs/faq-n-patches-offline

commit 27dc4831e88c952f3ea1f68b18434e16197fdfa9
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 13:50:27 2018 +0800

    Add XML Docs

commit baa8beb3824ed59fb50d269b9e13721fa9f5a04e
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 13:49:42 2018 +0800

    Add XML Docs

commit 089f97a0101a4ef9194f65b2d4435c0c69dfa87c
Author: Still Hsu <341464@gmail.com>
Date:   Sat May 5 13:43:04 2018 +0800

    Add details regarding userbot support

commit 9ddd70906aa01015fc76bc2401868e044c4c7cc1
Merge: f197174f 7cfed7ff
Author: Still Hsu <341464@gmail.com>
Date:   Fri May 4 09:37:04 2018 +0800

    Merge branch 'dev' into docs/faq-n-patches-offline

commit f197174fcc2df1e092b26cd2c8087b602ffb1960
Author: Still Hsu <341464@gmail.com>
Date:   Fri May 4 09:36:53 2018 +0800

    Fix embed docs consistency

commit 157acc46955d52809b959b52fb1a04404a6df3a5
Author: Still Hsu <341464@gmail.com>
Date:   Thu May 3 23:03:35 2018 +0800

    Add tag examples

commit 57ea571a8115f4b768c2dbbe1f2ef807b0cfd647
Author: Still Hsu <341464@gmail.com>
Date:   Thu May 3 22:48:33 2018 +0800

    Fix sample link & add missing pictures

commit 6769c37ad7e664845948c50ce403217a9951ef07
Author: Still Hsu <341464@gmail.com>
Date:   Thu May 3 22:39:26 2018 +0800

    Compress some assets & add OAuth2 URL generator

Commit:
Merge branch 'dev' into docs/faq-n-patches

Still34 commented:
Closed and will be re-opened when the PR is fully complete, as the current PR branch is obsolete anyways.
